test/unit/README.md

```diff
@@ -0,0 +1,14 @@
+# Extended unit tests
+
+Tests in this directory will be compiled into a single executable which links with
+the DMD frontend and runs all `unittest` blocks defined in this directory.
+
+## Purpose
+
+These tests are intended to test single components (e.g. the lexer) in isolation
+instead of relying on full end-to-end tests as done for `compilable`, ... .
```

Comment:
My intention was to be able to do what you described above, **and** replace `fail_compilation` and `compilable` with this test runner. For example [1], that would traditional be implemented as a `fail_compilation` test. Although it seems it's only me that has implement tests using this test runner.

[1] https://github.com/dlang/dmd/blob/master/test/unit/interfaces/check_implementations_20861.d

Reply:
Good to know but I don't think that replacing `fail_compilation` with `unit` tests is a worthwile endeavour. They require a lot of setup code, don't support `AUTO_UPDATE` and are prone for ICE's (because all tests are compiled into a single executable).

Also `fail_compilation` and `compilable` make only a small amount of the test suite's runtime, most of that is spent on `runnable` with `ARGS=-O -inline ...`